api/api/tests.py

```diff
@@ -1,5 +1,55 @@
+import functools
+import os
+import pytest
+import sys
+from django.conf import settings
+from django.urls.base import clear_url_caches
+from importlib import import_module, reload
 from rest_framework import status
 from rest_framework.test import APITestCase
+from unittest import mock
+
+
+def reload_root_urls(wrapped):
+
+    @functools.wraps(wrapped)
+    def wrapper(*args, **kwargs):
+        clear_url_caches()
+        if settings.ROOT_URLCONF in sys.modules:
+            reload(sys.modules['api.urls'])
+        else:
+            import_module(settings.ROOT_URLCONF)
+
+        return wrapped(*args, **kwargs)
+
+    return wrapper
+
+
+class AdminTest(APITestCase):
+    """By default, pytest doesn't load .env file"""
+
+    @pytest.mark.order(1)
+    @mock.patch.dict(os.environ, {'ENV': 'dev'})
+    @reload_root_urls
+    def testAdminEndpoint_inDevEnv_returns200Ok(self):
+        """Tests the /admin/ endpoint to return status code 200 when env is dev"""
+        url = '/admin/'
+        response = self.client.get(url, follow=True)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    @pytest.mark.order(2)
+    @reload_root_urls
+    def testAdminEndpoint_notInDevEnv_returns404(self):
+        """Tests the /admin/ endpoint to return status code 404 when env is not dev"""
+        url = '/admin/'
+        response = self.client.get(url, follow=True)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    @pytest.mark.order(3)
+    @reload_root_urls
+    def test_dummy_toResetRootUrls(self):
+        """Should always run at the end"""
```

Comment:
add the reason behind why it needs to run at the end. -> To reset the urlconf for default environment